docs/design-docs/multi-DM.md

```diff
@@ -66,3 +70,55 @@ behaviors, namely:
      LockingTransactions.client - to ensure that the lock identifier
      is consistent across all replicas.  This change should be
      straightforward.
+     
+## AtLeastOnceRPC + DHT + ConsensusRSM
+
+### Desired behavior
+
+1. Client creates a new instance or obtains a reference to an existing MicroService.
+2. Client makes a call to an application method in Microservice and gets the result back.
+3. Client application does not need to retry on network or other transient failures (as retries with timeout is automatically provided by AtLeastOnceRPC).
+4. The microservice is horizontally scalable (because sharding is provided by the DHT policy).
+5. The microservice is highly available, tolerating less than half the consensus replicas failing (as ConsensusRSM automatically, reliably and consistently replicates all method calls to all replicas of each shard).
+   
+### How it works under the hood
+
+1. Client creates an instance of a MicroService (create()).
+	1. Kernel invokes group.onCreate() on all DM's in the order of inner most DM to outer most DM.
+		1. ConsensusRSMPolicy.GroupPolicy.onCreate() creates 2f+1 replicas (by invoking 
+		replicate(), which in turn invokes addServer on all DM's).
+		1. DHTPolicy.GroupPolicy.onCreate() creates n shards, and configures the key range for each shard.
```

Comment:
nit: Looks to be numbering issue in this section

Reply:
This is intentional and following the previous convention. Use of the same numbers results in bullet marks which is the intention.